clusterloader2/testing/load/config.yaml

```diff
@@ -387,7 +389,12 @@ steps:
       {{range $EXEC_COMMAND}}
       - {{.}}
       {{end}}
+  - Identifier: WaitAfterExec
+    Method: Sleep
+    Params:
+        duration: {{$WAIT_AFTER_EXEC_DURATION}}
 {{end}}
+{{if not $EXIT_AFTER_EXEC}}
```

Comment:
having if spanning 500 lines seem like a bad idea TBH...

Reply:
@mm4tt

Reply:
Yeah, I agree it's far from perfection, but it seems to be the better than using skip-steps feature.

The goal is to reuse first part of the test to generate workload on cluster before running command and NOT running anything after the command that to remove potential impact of clusterloader on the operation started by the command.

An alternative I can see is to use --skip-steps feature for this, but this seems to be an error prone to me (we need to hardcopy list of all steps in the prowjob and adding any step to load test will require reflecting this in all prowjobs using this config). In the current approach, there is no risk that we will accidentally start running some step in our tests using exec step.

When I'm thinking about this now, I think we can have another solution: we can implement a "skip-all-steps-after--step" feature that will skip all 'phases' steps (leaving measurements) or removing all steps after the step with the given name.

Reply:
I am wondering if we shouldn't rethink format of our tests and maybe come up with some better idea than hardcoding steps/huge ifs/skipping some steps after.

Reply:
The issue with `--skip-all-steps-after-step` is that you need to skip some measurements (e.g. `WaitForControlledPods`) but you really want to execute some (e.g. collect on api-call-latency or resource usages). I agree with Marcel that solving that properly would probably require rethinking cl2 api and the structure of the test file. We could even kill two birds with a single stone and at the same time introduce "modularization" and refactor single huge config.yaml into multiple files. It would make the test easier to maintain / understand. But let's discuss this, it'd be a big change. I'd like to see a full design first before we dig into implementation.

Reply:
@mborsz @mm4tt @marseel - did you discuss that offline?

Being able to reference the other files (e.g. a single file defining a single step or a set of steps) would be useful. But that's a big thing to do, so I'm wondering if we don't need something more shorter-term.

Re -skip-all-steps-after-step would work - I like the idea modulo the fact that Matt described. That would work though if we would introduce the concept of deferred step, e.g. you do:
- step Start Metrics
- deferred-step Collect Metrics

Reply:
Briefly discussed offline - we will discuss that a bit more in the upcoming weeks and see how to attack this in a better way.